Announce and document RDS read replicas (#157)
cloud-gov/site: content/news/articles/2025-07-01-rds-read-replicas-available.md

---
layout: layouts/post
tags: news
title: Read replicas now available for RDS databases
date: 2025-07-01
excerpt: "Add a read replica for your RDS databases to handle read-only workloads"
---

> The Cloud.gov team is excited to announce that adding a read replica is now supported for your databases on the platform.

Reviewer: ```suggestion
The Cloud.gov team is excited to announce support for adding read replicas to your databases.
```

> [A read replica is a read-only copy of your database which automatically replicates the data in your primary database](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ReadRepl.html).
> 
> Read replicas are useful for offloading tasks that are read intensive while your primary database handles create, update, and deletion of records. Isolating read operations to the read replica reduces the load on your primary database and also avoids database locking issues between read and write operations.

Reviewer: ```suggestion
A read replica is a [read-only copy of your database which automatically replicates the data in your primary database](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ReadRepl.html). You can offload read-heavy tasks to a read replica while your primary database handles writes. Isolating read operations to the replica reduces the load and lock contention on your primary database.
```

> Examples of read intensive tasks that may good use cases for a read replica include (but are not limited to):

Reviewer: ```suggestion
Use cases for a read replica include:
```

- Generating reports from your data
- Serving data for API requests
- Performing analytics queries

> ## How to create a read replica

Reviewer: I think we should drop this entire section except lines 35-41, which explain the scope of the feature. The news post should be an announcement and link out to our service docs, which should have all of these instructions. Thoughts?

Author: https://github.com/cloud-gov/site/pull/157/commits/b9ca2b59ef79d5d283392845bd2f7c42e8fa38fd

To create a new database service with a read replica:

```shell
cf create-service aws-rds micro-psql-replica <your-service-name>
```

To update an existing database to a read replica plan:

```shell
cf update-service <your-service-name> -p micro-psql-replica
```

> > **Note:** `micro-psql-replica` is just an example of a service plan that supports database replicas. If you are updating an existing service to add a read replica, you should use the replica plan corresponding to your current database service (e.g. for databases currently using `medium-gp-psql*` plans, use the `medium-gp-psql-replica` plan).
> >
> > To see the list available of replica plans (which all end in `-replica`):
> >
> > ```shell
> > cf marketplace -e aws-rds
> > ```

Reviewer: Note, the suggestion isn't rendering correctly due to the fenced code block for `cf marketplace` — if you like this change, you'll want to commit it locally.

```suggestion
Read replicas are supported for all `aws-rds` plans. To add a read replica to an existing instance, use the replica plan corresponding to its current plan. For example, an instance using the `medium-gp-psql` plan should update to `medium-gp-psql-replica`.

To see the list available of replica plans (which all end in `-replica`):

```shell
cf marketplace -e aws-rds
```

```
